Crash on PatientChartActivity if there is no chart on the server

Steps to reproduce:

1. Set up a new local server. Do not apply a profile.
2. Connect tablet to server
3. Create a patient
4. Try to observe patient chart.
5. Patient chart crashes.

```
01-15 16:34:43.689 25041 25041 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{org.projectbuendia.client.dev/org.projectbuendia.client.ui.chart.PatientChartActivity}: java.lang.NullPointerException: Attempt to read from field 'java.lang.String org.projectbuendia.client.models.Chart.name' on a null object reference
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2379)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2442)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at android.app.ActivityThread.access$800(ActivityThread.java:156)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1351)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:211)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:5373)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Method.java:372)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1020)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:815)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to read from field 'java.lang.String org.projectbuendia.client.models.Chart.name' on a null object reference
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at org.projectbuendia.client.ui.chart.PatientChartActivity.initChartMenu(PatientChartActivity.java:296)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at org.projectbuendia.client.ui.chart.PatientChartActivity.onCreateImpl(PatientChartActivity.java:276)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at org.projectbuendia.client.ui.BaseLoggedInActivity.onCreate(BaseLoggedInActivity.java:80)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:5990)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2332)
01-15 16:34:43.689 25041 25041 E AndroidRuntime: 	... 10 more
```

Note also that this bug is being caught by `PatientChartControllerTest#testXformSubmitSucceeded_EventuallyHidesSubmissionDialog` and `PatientChartControllerTest#testPatientDetailsLoaded_SetsObservationsOnUi`, though the crash is reported as an `IndexOutOfBoundsException` instead of an NPE.

--- a/app/src/main/assets/no_chart.html
+++ b/app/src/main/assets/no_chart.html
@@ -12,6 +12,10 @@
 <br/><br/>
 <center>
     <h1>Invalid profile applied</h1>
+    <p>
+        Log in to the Server OpenMRS admin panel and choose
+        <i>Manage Buendia Profiles</i>.
+    </p>
 </center>
 </body>
 </html>

--- a/app/src/main/assets/no_chart.html
+++ b/app/src/main/assets/no_chart.html
@@ -1,0 +1,17 @@
+<!DOCTYPE html>
+<html>
+<title>Patient Chart</title>
+<meta charset="UTF-8">
+<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
+<style type="text/css">
+    h1 {
+        color: red;
+    }
+</style>
+<body>
+<br/><br/>
+<center>
+    <h1>Invalid profile applied</h1>
+</center>
+</body>
+</html>

--- a/app/src/main/java/org/projectbuendia/client/ui/chart/ChartRenderer.java
+++ b/app/src/main/java/org/projectbuendia/client/ui/chart/ChartRenderer.java
@@ -81,7 +81,10 @@ public class ChartRenderer {
                        List<Obs> observations, List<Order> orders,
                        LocalDate admissionDate, LocalDate firstSymptomsDate,
                        GridJsInterface controllerInterface) {
-
+        if (chart == null) {
+            renderError();
+            return;
+        }
         if ((observations.equals(mLastRenderedObs) && orders.equals(mLastRenderedOrders))
             && (lastChart.equals(chart.name))){
             return;  // nothing has changed; no need to render again
@@ -106,6 +109,10 @@ public class ChartRenderer {
 
         mLastRenderedObs = observations;
         mLastRenderedOrders = orders;
+    }
+
+    private void renderError() {
+        mView.loadUrl("file:///android_asset/no_chart.html");
     }
 
     class GridHtmlGenerator {

--- a/app/src/main/java/org/projectbuendia/client/ui/chart/ChartRenderer.java
+++ b/app/src/main/java/org/projectbuendia/client/ui/chart/ChartRenderer.java
@@ -82,7 +82,7 @@ public class ChartRenderer {
                        LocalDate admissionDate, LocalDate firstSymptomsDate,
                        GridJsInterface controllerInterface) {
         if (chart == null) {
-            renderError();
+            mView.loadUrl("file:///android_asset/no_chart.html");
             return;
         }
         if ((observations.equals(mLastRenderedObs) && orders.equals(mLastRenderedOrders))
@@ -109,10 +109,6 @@ public class ChartRenderer {
 
         mLastRenderedObs = observations;
         mLastRenderedOrders = orders;
-    }
-
-    private void renderError() {
-        mView.loadUrl("file:///android_asset/no_chart.html");
     }
 
     class GridHtmlGenerator {

